hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -91,6 +92,8 @@ public BaseCommitActionExecutor(HoodieEngineContext context, HoodieWriteConfig c
     // TODO : Remove this once we refactor and move out autoCommit method from here, since the TxnManager is held in {@link BaseHoodieWriteClient}.
     this.txnManager = new TransactionManager(config, table.getMetaClient().getFs());
     this.lastCompletedTxn = TransactionUtils.getLastCompletedTxnInstantAndMetadata(table.getMetaClient());
+    this.pendingRequestedInstants = TransactionUtils.getInflightInstants(table.getMetaClient());
```

Comment:
I think there is still a possibility of a race condition. The value for pendingRequestedInstants seen might be different from classes BaseHoodieWriteClient.java and BaseCommitActionExecutor.java.
@yihua What do you think?

Reply:
@suryaprasanna  @yihua  thanks for your remiding,  i think it's no problem, 
 the pendingInstants BaseCommitActionExecutor is used for autoCommit, 
1) pendingRequestedInstants is worked  together with lastCompletedTxn.
2) in BaseCommitActionExecutor we update  lastCompletedTxnand pendingRequestedInstants  together.
in autoCommit fuction,  we used updated   lastCompletedTxn and pendingRequestedInstants.  
when we check conflict  in resolveWriteConflictIfAny,  we combine  the pending Instants and    SimpleConcurrentFileWritesConflictResolutionStrategy.getCandidateInstants  together to check  conflict